### PR TITLE
Fix RNN

### PR DESCRIPTION
**Pull request by georgepar**
Fix bugs in RNN training

### CODE REVIEW

File: slp/modules/rnn.py [hunk 1]
```diff
@@ -9,29 +9,33 @@
...
+        if self.packed_sequence:
+            x, lengths = self.pack(x, lengths)
+        out, (hidden, cell) = self.rnn(x)
+        if self.packed_sequence:
             out = self.unpack(out, lengths)
         out = self.drop(out)
-        last_hidden = out[:, -1, :]
-        return out, last_hidden
+        # ->>> last_hidden = out[:, -1, :]
+        last_hidden = torch.cat((hidden[-2,:,:], hidden[-1,:,:]), dim=1) if self.bidirectional else hidden[-1,:,:]
```
Comment (georgepar) [inline slp/modules/rnn.py]: @ckarouzos Does this work for batch_first=True?

File: slp/modules/rnn.py [hunk 2]
```diff
@@ -60,9 +65,11 @@ def __init__(
 
     def forward(self, x, lengths):
         x = self.embed(x)
-        out, last_hidden = self.rnn(x, lengths)
+        out, last_hidden, hidden = self.rnn(x, lengths)
         if self.attention is not None:
-            out = self.attention(
-                out, attention_mask=pad_mask(lengths).to(self.device))
+            out, _ = self.attention(
+                out, attention_mask=pad_mask(lengths, device=self.device))
```
Comment (georgepar) [inline slp/modules/rnn.py]: @ckarouzos Have you tried using attention after the fix?

Comment (ckarouzos) [reply slp/modules/rnn.py]: Checked, working.

File: slp/modules/rnn.py (hunk 1 above)
Comment (ckarouzos) [reply slp/modules/rnn.py]: That is the case by default. 
However it's not tested for batch_first=False.

**Pull request merged**